### PR TITLE
Creating a user via graph Api

### PR DESCRIPTION
**Pull request by ScharfViktor**
### To unlock CI in ocis (webUITest failed), I need to switch the methods user creation  in midleware:

`Given user "Alice" has been created with default attributes and without skeleton files in the server`
and
`Given these users have been created with default attributes and without skeleton files in the server:`
to graphApi

#### How tested:
- start ocis with idm by default - branch https://github.com/owncloud/ocis/pull/3331
- start middleware with IDM variable: `BACKEND_HOST=https://localhost:9200 NODE_TLS_REJECT_UNAUTHORIZED=0 RUN_ON_OCIS=true TEST_WITH_GRAPH_API yarn start`
- init middleware: `curl -XPOST http://localhost:3000/init`
- try to create user "Alice": `curl -XPOST http://localhost:3000/execute -d '{"step": "Given user \"Alice\" has been created with default attributes and without skeleton files"}' -H "Content-Type: application/json"`

Result: Alice should be created: `curl  -k -u admin:admin https://localhost:9200/graph/v1.0/users/Alice`

### CODE REVIEW

**Review by individual-it — COMMENTED** [review]
looks mostly good to me
I've build a docker image with this code and pushed it to `arturneumann/owncloud-test-middleware:latest`
Here a PR that tries it https://github.com/owncloud/web/pull/6814

**Comment by kiranparajuli589** [comment]
I've rebased it and fixed the lint problems. we recently did lint updates in the repo, so you may have to do `rm -rf node_modules && yarn` to get the updated dependencies.

File: src/config.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const withHttp = (url) => (/^https?:\/\//i.test(url) ? url : `http://${url}`)
 
 const RUN_WITH_LDAP = !!process.env.RUN_WITH_LDAP
 const RUN_ON_OCIS = process.env.RUN_ON_OCIS === 'true'
+const RUN_WITH_IDM = process.env.RUN_WITH_IDM === 'true'
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply src/config.js:7]: we also have to document this new environment variable

[the remark this answers is not in the document]
Comment (ScharfViktor) [reply src/config.js:7]: done. I added a few words to the readme

File: src/helpers/graphHelper.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    password = password || userHelper.getPasswordForUser(user)
+
+    const body = JSON.stringify({
+        'displayName': displayName,
+        'mail': email,
+        'onPremisesSamAccountName': user,
+        'passwordProfile': { 'password': password }
+    })
+
+    console.log('step2 - create')
```
[the remark this answers is not in the document]
Comment (ScharfViktor) [reply src/helpers/graphHelper.js]: ops, I deleted it

**Comment by ScharfViktor** [comment]
> looks mostly good to me I've build a docker image with this code and pushed it to `arturneumann/owncloud-test-middleware:latest` Here a PR that tries it [owncloud/web#6814](https://github.com/owncloud/web/pull/6814)

thanks for advise))

**Comment by pascalwengerter** [comment]
I'll go ahead and merge this one and #109 and then release a new version of the middleware

**Pull request merged**